### PR TITLE
Code cleanup for dangling file pointer investigation

### PR DESCRIPTION
**Pull request by Anthchirp**
Investigation into https://github.com/dials/dials/pull/1603 found some superfluous programming motifs that made it unnecessarily difficult to figure out what is going on.

### CODE REVIEW

File: datablock.py [hunk 1]
```diff
@@ -1502,7 +1501,7 @@ def from_dict(obj, check_format=True, directory=None):
     def from_json(string, check_format=True, directory=None):
         """Decode a datablock from JSON string."""
         return DataBlockFactory.from_dict(
-            json.loads(string, object_hook=_decode_dict),
+            json.loads(string),
```
Comment (graeme-winter) [inline datablock.py:1504]: was the `object_hook` stuff really useless?

File: model/experiment_list.py [hunk 2]
```diff
@@ -1,14 +1,9 @@
-from __future__ import absolute_import, division, print_function
```
Comment (graeme-winter) [inline model/experiment_list.py:1]: 🙂

File: serialize/load.py [hunk 3]
```diff
@@ -1,48 +1,11 @@
...
 import os
-
-import six
+import warnings
 
 from dxtbx.model.crystal import CrystalFactory
 from dxtbx.serialize.imageset import imageset_from_dict
 
 
-def _decode_list(data):
```
Comment (graeme-winter) [inline serialize/load.py:12]: <oh> this is only useless now we are in _the future_ 👍

**Review by graeme-winter — APPROVED** [review]
Nice clean up 👍

Tests pass, code makes sense, expect there is more of this nonsense to be found but this still helps thank you

File: datablock.py (hunk 1 above)
Comment (Anthchirp) [reply datablock.py:1504]: on Python 3 it was a no-op

**Pull request merged**